### ISSUE TITLE
Path as argument is broken?

### ISSUE THREAD

**Issue by gforcada** [reinout/z3c.dependencychecker#53]
I'm having some trouble trying to understand the idea with the ``determine_path`` method: https://github.com/reinout/z3c.dependencychecker/blob/master/z3c/dependencychecker/dependencychecker.py#L577

If I pass another folder, i.e.
```
git clone z3c.dependencychecker (virtualenv and what not)
git clone plone.scale

cd z3c.dependencychecker
./bin/dependencychecker -v ../plone.api
```

With that I get the following output:
```
DEBUG: Looking in directory ../plone.scale, passed on the command line
DEBUG: No imported modules found.
DEBUG: No imported modules found.
DEBUG: All found regular imported packages: []
DEBUG: All found regular imported test packages: []
DEBUG: Appended ourselves (z3c.dependencychecker) to the required packages.
DEBUG: All found zcml-related packages: []
DEBUG: All found zcml-related test packages: []
DEBUG: All found FTI-related packages: []
DEBUG: All found FTI-related test packages: []
DEBUG: All found django_settings-related packages: []
DEBUG: All found django_settings-related test packages: []
DEBUG: All found generic_setup-related packages: []
DEBUG: All found generic_setup-related test packages: []
Unneeded test requirements
==========================
     pytest

Note: requirements are taken from the egginfo dir, so you need
to re-run buildout (or setup.py or whatever) for changes in
setup.py to have effect.
```

As you can see on the output it complains about ``pytest`` (which is not in plone.scale but rather on z3c.dependencychecker) and it also adds ``z3c.depenencychecker`` rather than plone.scale as the own package:
```
DEBUG: Appended ourselves (z3c.dependencychecker) to the required packages.
```

This last error I think comes from ``name_from_setup`` as there we call ``python setup.py --name`` without taking into account the, possibly, given path as a command line argument.

The former error, pytest being reported, probably is somewhat related? We analyze ``requires.txt`` from where we call rather than from where the argument was given?

Bottom line is, what was the purpose here, to do:
-  ``./bin/dependencychecker ../plone.scale`` or
- ``cd ../plone.scale && ../z3c.dependencychecker/bin/dependencychecker``

The later works, the former does not.

IMHO I would favor the former approach (pass the root path of the source checkout where setup.py should be found) and if not given assume ``os.getcwd()`` and march forward with the analysis.

@reinout @mauritsvanrees ideas, comments? 😃

**Comment by gforcada** [comment]
@reinout given that you did not negate the approach I took on the current rewrite, I guess you agree on the first approach.

I will close this then.
